### PR TITLE
BACKUP-123 : 5.7 backup plugin fails to compile.

### PR DESCRIPTION
**Pull request by george-lorch**
- Most compilation errors were due to my_malloc, my_realloc, my_strdup requiring a PSI_memory_key which we do not yet have. Added argument to pass a static mem_key to all places where needed and later we can add the proper PSI instrumentation and key management.
- 5.7 changes the interface to get_system_var and requires a Parse_context, which is something that must be provided (not NULL) and something that we do not have available to us in the backup pligin. We now re-implement some of what get_system_var does to get at these variables within find_plug_in_sys_var.

### CODE REVIEW

File: plugin/tokudb-backup-plugin/tokudb_backup.cc [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #endif
 
 static char *tokudb_backup_plugin_version;
+static PSI_memory_key tokudb_backup_mem_key = 0;
```
Comment (laurynas-biveinis) [inline plugin/tokudb-backup-plugin/tokudb_backup.cc:35]: I'm not sure I like this approach that creates what appears to be a memory key, yet is not a memory key. Readers at my_*alloc call will be led to thinking that the calls are annotated. Either go all the way already (it's ~5 more lines of code + unknown amount of MTR re-records for PFS), either just use PSI_UNINSTRUMENTED (sic?)

File: plugin/tokudb-backup-plugin/tokudb_backup.cc [hunk 2]
```diff
@@ -134,7 +135,11 @@ static int tokudb_backup_progress_fun(float progress, const char *progress_strin
     // set thd proc info
     thd_proc_info(be->_thd, "");
     size_t len = 100 + strlen(progress_string);
-    be->_the_string = (char *) my_realloc(be->_the_string, len, MYF(MY_FAE+MY_ALLOW_ZERO_PTR));
+    be->_the_string =
+        (char*)my_realloc(tokudb_backup_mem_key,
```
Comment (laurynas-biveinis) [inline plugin/tokudb-backup-plugin/tokudb_backup.cc]: Should we be using static_cast<> for new and touched code?

**Comment by george-lorch** [comment]
Well, I did this to seed the creation of a real key for this plugin which I already registered a task earlier today https://tokutek.atlassian.net/browse/BACKUP-124 to fill in the blanks. I needed to get something compiling for testing and quickly and this all did the trick. It follows the changes you made to tokudb_memory.h (passing in 0 for the PSI key) but this just went a bit further to define a key for everything to use now even if it is just 0. Honestly, I was more concerned about the changes needed to find_plug_in_sys_var as I did not like needing to replicate server internals behavior and was hoping that there was a better way.

File: plugin/tokudb-backup-plugin/tokudb_backup.cc [hunk 3]
```diff
@@ -421,21 +436,43 @@ class source_dirs {
...
-        if (item) {
-            String scratch;
-            String * str = item->val_str(&scratch);
-            if (str) {
-                result = my_strdup(str->ptr(), MYF(MY_FAE));
-            }
+        LEX_STRING component_name = name_to_find.lex_string();
+
+
+        // 5.7 change the interface to get_system_var and requires a
```
Comment (laurynas-biveinis) [inline plugin/tokudb-backup-plugin/tokudb_backup.cc:454]: why this was get_system_var in the first place? do_deprecated_warning seems redundant? the caller appears to read two string vars and that's all. Why not just find_sys_var, followed by var->value_ptr ?

Comment (george-lorch) [inline plugin/tokudb-backup-plugin/tokudb_backup.cc:454]: Good question, you would probably have to ask the original implementors. Based on some other comments in the code, it _might_ have been that this was the most universal approach for covering all variants and versions supported.

**Comment by laurynas-biveinis** [comment]
OK, Current no-op key is fine if BACKUP-124 will be implemented ASAP

**Comment by george-lorch** [comment]
OK, so I added a comment to the declaration of the mem_key indicating that it is only a temporary place holder, fixed up a few of the casts on touching code to be static_cast and removed some redundancy/not needed calls from find_pug_in_sys_var and retested for what that is worth :) I am going to go ahead and merge so we can try to begin testing on this and new issues can be brought up against BACKUP.

**Comment by george-lorch** [comment]
jenkins link showing successful builds now of backup plugin http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/78/

**Pull request merged**